specification/apimanagement/control-plane/Microsoft.ApiManagement/preview/2017-03-01/apimemailtemplate.json

```diff
@@ -0,0 +1,407 @@
+        "x-ms-examples": {
+          "ApiManagementListTemplates": {
+            "$ref": "./examples/ApiManagementListTemplates.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "./apimanagement.json#/parameters/TopQueryParameter"
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/SkipQueryParameter"
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "A Collection of the Email Template for the specified API Management service instance.",
+            "schema": {
+              "$ref": "#/definitions/EmailTemplateCollection"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/templates/{templateName}": {
+      "get": {
+        "tags": [
+          "EmailTemplates"
+        ],
+        "operationId": "EmailTemplate_Get",
+        "description": "Gets the details of the email template specified by its identifier.",
+        "x-ms-examples": {
+          "ApiManagementGetTemplates": {
+            "$ref": "./examples/ApiManagementGetTemplates.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/TemplateNameParameter"
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The response body contains the specified Email template.",
+            "schema": {
+              "$ref": "#/definitions/EmailTemplateContract"
+            },
+            "headers": {
+              "ETag": {
+                "description": "Current entity state version. Should be treated as opaque and used to make conditional HTTP requests.",
+                "type": "string"
+              }
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "./apimanagement.json#/definitions/ErrorResponse"
+            }
+          }
+        }
+      },
+      "put": {
+        "tags": [
+          "EmailTemplates"
+        ],
+        "operationId": "EmailTemplate_CreateOrUpdate",
+        "description": "Updates an Email Template.",
+        "x-ms-examples": {
+          "ApiManagementPutTemplates": {
+            "$ref": "./examples/ApiManagementPutTemplates.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/TemplateNameParameter"
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/EmailTemplateUpdateParameters"
```

Comment:
>EmailTemplateUpdateParameters [](start = 37, length = 29)

Should this be `EmailTemplateContract` ?

Reply:
It is EmailTemplateUpdateParameters, as by default we have templates for all emails, so customer essentially always updates it. Also none are required params